controllers/pkg/reconcilers/repository/reconciler.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/go-logr/logr"
+	ctrlconfig "github.com/henderiw-nephio/nephio-controllers/controllers/config"
+	"github.com/henderiw-nephio/nephio-controllers/pkg/applicator"
+	infrav1alpha1 "github.com/nephio-project/api/infra/v1alpha1"
+	"github.com/nephio-project/nephio/controllers/pkg/giteaclient"
+	"github.com/nokia/k8s-ipam/pkg/meta"
+	"github.com/nokia/k8s-ipam/pkg/resource"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+/*
+func init() {
+	controllers.Register("repositories", &reconciler{})
+}
+*/
+
+const (
+	finalizer = "infra.nephio.org/finalizer"
+	// errors
+	errGetCr        = "cannot get cr"
+	errUpdateStatus = "cannot update status"
+)
+
+//+kubebuilder:rbac:groups=infra.nephio.org,resources=repositories,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=infra.nephio.org,resources=repositories/status,verbs=get;update;patch
+
+// Setup sets up the controller with the Manager.
+func (r *Reconciler) Setup(mgr ctrl.Manager, cfg *ctrlconfig.ControllerConfig) (map[schema.GroupVersionKind]chan event.GenericEvent, error) {
+	if err := infrav1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
+		return nil, err
+	}
+
+	r.APIPatchingApplicator = applicator.NewAPIPatchingApplicator(mgr.GetClient())
+	r.giteaClient = cfg.GiteaClient
+	r.finalizer = resource.NewAPIFinalizer(mgr.GetClient(), finalizer)
+
+	return nil, ctrl.NewControllerManagedBy(mgr).
+		Named("RepositoryController").
+		For(&infrav1alpha1.Repository{}).
+		Complete(r)
+}
+
+type Reconciler struct {
+	applicator.APIPatchingApplicator
+	giteaClient giteaclient.GiteaClient
+	finalizer   *resource.APIFinalizer
+
+	l logr.Logger
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.l = log.FromContext(ctx)
+	r.l.Info("reconcile", "req", req)
+
+	cr := &infrav1alpha1.Repository{}
+	if err := r.Get(ctx, req.NamespacedName, cr); err != nil {
+		// if the resource no longer exists the reconcile loop is done
+		if resource.IgnoreNotFound(err) != nil {
+			r.l.Error(err, "cannot get resource")
+			return ctrl.Result{}, errors.Wrap(resource.IgnoreNotFound(err), "cannot get resource")
```

Comment:
can we just use err here instead of resource.IgnoreNotFound(err)?

Reply:
I believe there is a different behaviour when the resource is not there with not found versus resource cannot be retrieved. In the ignore not found case it stops. in the other case it triggers a new reconcile.